noir/noir-repo/aztec_macros/src/transforms/compute_note_hash_and_nullifier.rs

```diff
@@ -59,13 +62,86 @@ pub fn inject_compute_note_hash_and_nullifier(
             return Ok(());
         }
 
+        let traits: Vec<_> = collect_traits(context);
+
         // In order to implement compute_note_hash_and_nullifier, we need to know all of the different note types the
         // contract might use. These are the types that are marked as #[aztec(note)].
-        let note_types =
-            fetch_notes(context).iter().map(|(path, _)| path.to_string()).collect::<Vec<_>>();
+        let mut note_types = vec![];
+        let mut note_lengths = vec![];
```

Comment:
If you don't want to create a new struct, maybe at least make a single vector of tuples since these will share indices?

Reply:
Done, but I have to map it afterwards to get length and types...maybe it's more readable like this, not sure.